### ISSUE TITLE
SERVICE_START audit messages logged just before SERVICE_STOP when shutting down

### ISSUE THREAD

**Issue by bigon** [systemd/systemd#9962]
Hi,

In some occasions, I can see SERVICE_START audit messages being logged just before SERVICE_STOP when shutting down.

Example:

```
aoû 28 11:43:12 valinor thermald[789]: Terminating ...
aoû 28 11:43:13 valinor thermald[789]: terminating on user request ..
aoû 28 11:43:14 valinor audit[1]: SERVICE_START pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=thermald comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
aoû 28 11:43:14 valinor audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 subj=system_u:system_r:init_t:s0 msg='unit=thermald comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
```

**systemd version the issue has been seen with**
> 239

**Used distribution**
 > Debian

**Expected behaviour you didn't see**
 > Only SERVICE_STOP audit messages are logged for stopped services

**Unexpected behaviour you saw**
 > Both SERVICE_START and then immediately after SERVICE_STOP audit messages are logged when the service is stopped

**Steps to reproduce the problem**

**Comment by bigon** [comment]
I can reproduce that only if `systemctl daemon-reload` or `systemctl daemon-reexec` has been called at least once.

**Comment by poettering** [comment]
Ah, this is a bug, we are currently not serializing across a daemon-reload/daemon-reexec the per-unit "in_audit" boolean that tracks whether we already sent out SERVICE_START. Hence after each cycle we'll assume again that the service has not been passed to audit yet, and thus you see this behaviour.

Fixing this should be easy, just needs the missing serialization.

**Comment by poettering** [comment]
Fix waiting in #10324
